### ISSUE TITLE
Feature: store last processed event as an annotation

### ISSUE THREAD

**Issue by lblackstone** [pulumi/pulumi-kubernetes#615]
The provider's await logic uses k8s Events to determine readiness for k8s resources. There are at least a couple cases where it would be useful to have a record of the Event that was last used to determine readiness:
1. Debugging edge cases - could run the recorded Event through the await logic to find logical errors
1. Avoid processing duplicated Events -- see https://github.com/pulumi/pulumi-kubernetes/issues/611#issuecomment-506795445

I propose that we store the last processed Event as an annotation on the resource.

**Comment by hausdorff** [comment]
I suggested an immediate-term mitigation for that bug [in the thread](https://github.com/pulumi/pulumi-kubernetes/issues/611#issuecomment-508274635) -- I recommend we proceed ASAP with it, as I believe it will completely mitigate (2), and this is a critical bug.

I'm not too sure about storing the last `v1.WatchEvent` though. We are going to be storing the last applied configuration in soon, and with this that would effectively triple the size of ~every kube object. Also, it complicates the mechanics of doing updates, will probably confuse users, and so on.

If we thought that this would _really_ help debug edge cases, that could be compelling, but I can't think of a case that justifies doubling the size of every resource definition.

Thoughts?
